### ISSUE TITLE
compiler version printing

### ISSUE THREAD

**Issue by Drup** [ocaml/opam#1946]
When I tried to installed codoc, I got the message: 
`
opam-doc-base is not available because it requires OCaml = 4.02.0.
`

When I did `opam show --raw opam-doc-base` I got:
```
opam-version: "1"
name: "opam-doc-base"
version: "0.1.3"
maintainer: "lpw25@cl.cam.ac.uk"
authors: "Leo White <lpw25@cl.cam.ac.uk>"
license: "ISC"
build: [
  ["assemblage" "setup" "--dumpast=false"]
  [make]
  [make "install"]
]
remove: ["ocamlfind" "remove" "opam-doc-base"]
depends: [
  "cmdliner"
  "xmlm"
  "opam-lib"
  "opam-units"
  "ocamlgraph"
  "ocamlfind"
  "assemblage"
]
ocaml-version: [= "4.02.0"]
```

However, as you can see [here](https://github.com/ocaml/platform-dev/blob/master/packages/opam-doc-base/opam-doc-base.0.1.3/opam). The compiler version is not right. It seems that `+doc` is not printed.

**Comment by AltGr** [comment]
Can you check with `cat $(opam show --where opam-doc-base)` that it's really OPAM adding a difference ?
Thanks

**Comment by AltGr** [comment]
Ah nevermind, indeed we do have:
https://github.com/ocaml/opam/blob/869d316c3c2c29f40e43113ab55f996ef9723779/src/core/opamCompiler.ml#L27

which means everything after `+` is always ignored. Hum...

**Comment by AltGr** [comment]
Remember though, `ocaml-version` is the base OCaml version, so while there is a bug in the parser/printer, what you want is `available: [  compiler = "4.02.0+doc" ]`

See `opam config list` for a reminder of the difference between `ocaml-version`, `compiler` and `switch`.

**Comment by Drup** [comment]
> Remember though, ocaml-version is the base OCaml version, so while there is a bug in the parser/printer, what you want is available: [ compiler = "4.02.0+doc" ]

cc @dsheets @samoht 
I'm not responsible of that. :)

**Comment by dsheets** [comment]
Thanks, @Drup!
